code/datums/elements/recharging_attacks.dm

```diff
@@ -0,0 +1,58 @@
+/// Reduces the cooldown of a given action upon landing attacks, critting, or killing mobs.
+/datum/element/recharging_attacks
```

Comment:
This cannot be an element because you are passing in a separate instance of `datum/action/cooldown` every time, so every mob will have a new instance of this element anyway, this should be a component